File: Structural/Proxy/Cpp/RealVideoService.cpp
// RealVideoService.cpp

#include "RealVideoService.h"

void RealVideoService::loadVideo(const std::string& videoName) {
  std::cout << "Loading video: " << videoName << std::endl;
  std::this_thread::sleep_for(std::chrono::seconds(2)); // Simulating video loading delay
  std::cout << "Playing video: " << videoName << std::endl;
}

File: Structural/Proxy/CSharp/RealVideoService.cs
// RealVideoService.cs

using System;
using System.Threading;

public class RealVideoService : IVideoService {
  public void LoadVideo(string videoName) {
    Console.WriteLine($"Loading video: {videoName}");
    Thread.Sleep(2000); // Simulating video loading delay
    Console.WriteLine($"Playing video: {videoName}");
  }
}
